include/openenclave/bits/sgx/sgxtypes.h

```diff
@@ -668,6 +668,27 @@ OE_PACK_END
 
 OE_CHECK_SIZE(sizeof(sgx_quote_t), 436);
 
+/*
+**==============================================================================
+**
+** sgx_ql_attestation_algorithm_id_t
+**
+**==============================================================================
+*/
+// Enumerates the different attestation key algorithms
+// For the sign_type field in sgx_quote_t
+typedef enum
+{
+    SGX_QL_ALG_EPID = 0,       // EPID 2.0 - Anonymous: EPID unlinkable
+    SGX_QL_ALG_RESERVED_1 = 1, // Reserved: EPID linkable
+    SGX_QL_ALG_ECDSA_P256 = 2, // ECDSA-256-with-P-256 curve
+    SGX_QL_ALG_ECDSA_P384 = 3, // ECDSA-384-with-P-384 curve (not supported)
+    SGX_QL_ALG_MAX = 4
+} sgx_ql_attestation_algorithm_id_t;
```

Comment:
Since this is a public header, shouldn't these start with oe_ in front?
https://github.com/dthaler/openenclave/blob/master/docs/Community/DevelopmentGuide.md says

> Prefix Open Enclave specific names in the global namespace with oe_ (e.g. oe_result_t, oe_call_enclave).

If these are not OE-specific names, what happens if the app includes this header and (say) some Intel header it also uses for something else.  There can be compiler errors due to name collision.
```suggestion
} oe_sgx_ql_attestation_algorithm_id_t;
```

Same question on line 690, etc.

Reply:
This header file, `sgxtypes.h`, is special. It replicates the definitions of SGX-specific data types, using the same names as that in the Intel SGX SDK, but redefines some of the internal fields that are not touched in the OE SDK code.

It is a public header  file, used in tests and a number of samples. But most of the structures / fields look like are internal to the OE SDK code.

I think eventually, all SGX-specific definitions should be made internal, and this header file be moved to internal. When the SGX SDK header files package is available as a dependency, this file can be updated to include the SGX SDK header files directly.

Reply:
Ok, opened issue #3447 to track this discussion.